keeper/msg_server.go

```diff
@@ -446,3 +421,39 @@ func (k *Keeper) claimExternalYieldHyperlane(ctx context.Context) error {
 
 	return nil
 }
+
+func (k *Keeper) handleVaultsYield(ctx context.Context, index int64) error {
+	// Claim the yield of the Flexible vault.
+	flexibleYield, err := k.claimModuleYield(ctx, vaults.FlexibleVaultAddress)
+	if err != nil {
+		return err
+	}
+
+	// Claim the yield of the Staked vault and redirect it to the Flexible vault.
+	stakedYield, err := k.claimStakedVaultYield(ctx)
+	if err != nil {
+		return err
+	}
+
+	// get the current Flexible total principal.
+	totalFlexiblePrincipal := math.ZeroInt()
+	if has, _ := k.VaultsTotalFlexiblePrincipal.Has(ctx); has {
+		current, err := k.VaultsTotalFlexiblePrincipal.Get(ctx)
+		if err != nil {
+			return err
+		}
+		totalFlexiblePrincipal = totalFlexiblePrincipal.Add(current)
+	}
```

Comment:
I know this logic was just refactored on this PR but just wondering if I understand the use of the collections module correctly here because it seems redundant to me:

So `.Has(ctx)` is checking if the value is set? But if it's not set, the `.Get(ctx)` would just return `math.Int{}` which is zero anyways? And then we're adding the returned value to a new `math.ZeroInt()`, so I think this whole logic could be simplified to


```suggestion
	// get the current Flexible total principal.
	totalFlexiblePrincipal, err := k.VaultsTotalFlexiblePrincipal.Get(ctx)
	if err != nil {
		return err
	}
```